modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor_classic.jsp

```diff
@@ -43,9 +45,15 @@ Map<String, Object> data = HashMapBuilder.<String, Object>put(
 	"name", HtmlUtil.escapeAttribute(name)
 ).put(
 	"onChangeMethodName", HtmlUtil.escapeJS(onChangeMethod)
+).put(
+	"title", LanguageUtil.get(request, placeholder)
 ).build();
 %>
 
+<liferay-editor:resources
+	editorName="<%= editorName %>"
```

Comment:
I know we're fans of DRY... but maybe we'd rather copy the things we need from there either here or even better inside the React component?

Doing this makes me worry that:
- Using from `jsp` is safe
- Using directly from `js` is unsafe

Reply:
The main problem is how to load `ckeditor.js` in a React + SPA context. In [resources.jsp](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/resources.jsp) , we are doing it with 
```js
<script data-senna-track="temporary" src="<%= HtmlUtil.escapeAttribute(PortalUtil.getStaticResourceURL(request, themeDisplay.getCDNHost() + PortalWebResourcesUtil.getContextPath(PortalWebResourceConstants.RESOURCE_TYPE_EDITOR_CKEDITOR) + "/ckeditor/ckeditor.js", javaScriptLastModified)) %>" type="text/javascript"></script>
```
I am not sure what is the best way to do this, I'll try tinkering around! I didn't see any way in CKEditor API to do a manual initialization, `CKEDITOR`  object is the base API entry point, [initialized eagerly](https://github.com/ckeditor/ckeditor4/blob/major/core/ckeditor_base.js).

Reply:
But we pass a path to the React component so it loads it automatically and not through the `resources` direct `script`.

Reply:
This works initially, after page refresh. But not when navigating to the page.

To test this, just delete
```js
<liferay-editor:resources
	editorName="<%= editorName %>"
/>
```
Then
1. Open edit message page. Refresh page. Editor works.
1. Cancel.
1. Edit message. Editor breaks because `CKEDITOR` is undefined.

Reply:
Yeah, what I mean is... can we add something to `<ClassicEditor>` to account for this rather than adding `<liferay-editor:resources >`?

Reply:
I'm working on it, but it's probably not gonna be pretty.